clio-server/src/main/scala/org/broadinstitute/clio/server/webservice/BamWebService.scala

```diff
@@ -0,0 +1,29 @@
+package org.broadinstitute.clio.server.webservice
+
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server._
+import org.broadinstitute.clio.server.service.IndexService
+import org.broadinstitute.clio.transfer.model.BamIndex
+import org.broadinstitute.clio.transfer.model.bam.BamKey
+import org.broadinstitute.clio.util.model.{DataType, Location}
+
+class BamWebService(bamService: IndexService[BamIndex.type])
+    extends IndexWebService(bamService) {
+
+  private[webservice] val pathPrefixKey: Directive1[BamKey] = {
+    for {
+      location <- pathPrefix(Location.namesToValuesMap)
```

Comment:
We had talked about calling this `processingLocation` instead of `location`. Did this not turn out to be a good idea?

Reply:
`location` is a required key by Clio. We're going to keep `processingLocation` info in the on-prem database